### PR TITLE
Model refactoring

### PR DESCRIPTION
**Pull request by lechup**
* enable gis on models
* added gis fields
* cleanup of model structure

---
This change is [Reviewable](https://reviewable.io/reviews/enviromonitor/enviromonitorweb/30)

### CODE REVIEW

File: api/models.py [hunk 1]
```diff
@@ -1,29 +1,145 @@
...
+class LocationBase(models.Model):
+    """
+    Base Model for location features.
+    """
+
+    position = gis_models.PointField(help_text='Exact position on map.', default=None, null=True)
+    country = models.CharField(max_length=255, default='')
+    state = models.CharField(max_length=255, default='')
+    county = models.CharField(max_length=255, default='')
+    community = models.CharField(max_length=255, default='')
```
Comment (lechup) [inline api/models.py]: @zen what is rationale behind community field? Other fields are quite self explanatory...

**Comment by zen** [comment]
Review status: 0 of 11 files reviewed at latest revision, 1 unresolved discussion.

---

*[api/models.py, line 103 at r1](https://reviewable.io:443/reviews/enviromonitor/enviromonitorweb/30#-KaqOYG3B_FIE7z-9-p2-r1-103:-KarCEpXAzJAZLicPqBr:b-aztz2x) ([raw file](https://github.com/enviromonitor/enviromonitorweb/blob/138896fb34c097c0926a8553c11fc5ddf8388d3f/api/models.py#L103)):*
<details><summary><i>Previously, lechup (Leszek Piątek) wrote…</i></summary><blockquote>

@zen what is rationale behind community field? Other fields are quite self explanatory...
</blockquote></details>

Community - is just abstraction for lowest level of geographical location. It can be "district" or other location. Maybe it's not that clear as I think of it now.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/enviromonitor/enviromonitorweb/30)*

**Comment by lechup** [comment]
Review status: 0 of 11 files reviewed at latest revision, 1 unresolved discussion.

---

*[api/models.py, line 103 at r1](https://reviewable.io:443/reviews/enviromonitor/enviromonitorweb/30#-KaqOYG3B_FIE7z-9-p2-r1-103:-KaukHeC2Z8RDDAC-frR:b-5nh0ne) ([raw file](https://github.com/enviromonitor/enviromonitorweb/blob/138896fb34c097c0926a8553c11fc5ddf8388d3f/api/models.py#L103)):*
<details><summary><i>Previously, zen (Tomasz Napierala) wrote…</i></summary><blockquote>

Community - is just abstraction for lowest level of geographical location. It can be "district" or other location. Maybe it's not that clear as I think of it now.
</blockquote></details>

But we have separate `district` field. Can I factor it out for now? For me ForeignKey to Project will do the job - it will link Station with Project/community?

What do You think?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/enviromonitor/enviromonitorweb/30)*

**Comment by grillazz** [comment]
Reviewed 10 of 11 files at r1.
Review status: 10 of 11 files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/enviromonitor/enviromonitorweb/30)*

**Comment by lechup** [comment]
Review status: 10 of 11 files reviewed at latest revision, 1 unresolved discussion.

---

*[api/models.py, line 103 at r1](https://reviewable.io:443/reviews/enviromonitor/enviromonitorweb/30#-KaqOYG3B_FIE7z-9-p2-r1-103:-Kavs-7WuPbS4Y5KnJvH:b-8naja7) ([raw file](https://github.com/enviromonitor/enviromonitorweb/blob/138896fb34c097c0926a8553c11fc5ddf8388d3f/api/models.py#L103)):*
<details><summary><i>Previously, lechup (Leszek Piątek) wrote…</i></summary><blockquote>

But we have separate `district` field. Can I factor it out for now? For me ForeignKey to Project will do the job - it will link Station with Project/community?

What do You think?
</blockquote></details>

Ok, so community is part of location and we need to leave it - for PL we need it to save info about `gmina`.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/enviromonitor/enviromonitorweb/30)*

**Comment by lechup** [comment]
Reviewed 1 of 12 files at r1, 3 of 3 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/enviromonitor/enviromonitorweb/30)*

**Pull request merged**